data/models/auth-model.js

```diff
@@ -0,0 +1,35 @@
+const db = require('../../utilities/dbConfig');
+const bcrypt = require('bcryptjs');
+
+const checkUsername = async username => {
+  if (
+    await db('users')
+      .where({ username: username.toLowerCase() })
+      .first()
+  ) {
+    return 'taken';
+  } else {
+    return 'available';
+  }
+};
+
+const registerUser = async user => {
+  const password = bcrypt.hashSync(user.password, 8);
+  return db('users').insert({
+    username: user.username.toLowerCase(),
```

Comment:
I didn't comment on this in the week's earlier projects, but while this isn't a terrible idea when validating the data (if you want the usernames to basically be case sensitive when signing on), is it entirely necessary to set it all to lowercase when inserting into the database? you'd have no way then of ever checking exactly how the user submitted their data when registering, then, and they might not be expecting a normalized version of their username if was ever reflected anywhere in the app's front end

Reply:
I was trying to follow Google's best practices, but I see that the user would not expect "Hello, chris" if they entered their username as "Chris". I think I can fix this.